EventFilter/HcalRawToDigi/plugins/HcalDigiToRawuHTR.cc

```diff
@@ -0,0 +1,284 @@
+  std::string electronicsMapLabel_;
+
+  edm::EDGetTokenT<HcalDataFrameContainer<QIE10DataFrame> > tok_QIE10DigiCollection_;
+  edm::Handle<QIE10DigiCollection> qie10DigiCollection;
+  edm::EDGetTokenT<HcalDataFrameContainer<QIE11DataFrame> > tok_QIE11DigiCollection_;
+  edm::Handle<QIE11DigiCollection> qie11DigiCollection;
+  edm::EDGetTokenT<HBHEDigiCollection> tok_HBHEDigiCollection_;
+  edm::Handle<HBHEDigiCollection> hbheDigiCollection;
+  edm::EDGetTokenT<HFDigiCollection> tok_HFDigiCollection_;
+  edm::Handle<HFDigiCollection> hfDigiCollection;
+  edm::EDGetTokenT<HcalTrigPrimDigiCollection> tok_TPDigiCollection_;
+  edm::Handle<HcalTrigPrimDigiCollection> tpDigiCollection;
+
+  edm::InputTag qie10Tag_, qie11Tag_, hbheqie8Tag_, hfqie8Tag_, trigTag_;
+};
+
+HcalDigiToRawuHTR::HcalDigiToRawuHTR(const edm::ParameterSet& iConfig) :
+  _verbosity(iConfig.getUntrackedParameter<int>("Verbosity", 0)),
+  electronicsMapLabel_(iConfig.getParameter<std::string>("ElectronicsMap")),
+  qie10Tag_(iConfig.getParameter<edm::InputTag>("QIE10")),
+  qie11Tag_(iConfig.getParameter<edm::InputTag>("QIE11")),
+  hbheqie8Tag_(iConfig.getParameter<edm::InputTag>("HBHEqie8")),
+  hfqie8Tag_(iConfig.getParameter<edm::InputTag>("HFqie8")),
+  trigTag_(iConfig.getParameter<edm::InputTag>("TP"))
+{
+  produces<FEDRawDataCollection>("");
+  tok_QIE10DigiCollection_ = consumes<HcalDataFrameContainer<QIE10DataFrame> >(qie10Tag_);
+  tok_QIE11DigiCollection_ = consumes<HcalDataFrameContainer<QIE11DataFrame> >(qie11Tag_);
+  tok_HBHEDigiCollection_ = consumes<HBHEDigiCollection >(hbheqie8Tag_);
+  tok_HFDigiCollection_ = consumes<HFDigiCollection>(hfqie8Tag_);
+  tok_TPDigiCollection_ = consumes<HcalTrigPrimDigiCollection>(trigTag_);
+}
+
+HcalDigiToRawuHTR::~HcalDigiToRawuHTR(){}
+
+void HcalDigiToRawuHTR::produce(edm::Event& iEvent, const edm::EventSetup& iSetup){
+
+  using namespace edm;
+
+  edm::ESHandle<HcalDbService> pSetup;
+  iSetup.get<HcalDbRecord>().get( pSetup );
+  edm::ESHandle<HcalElectronicsMap> item;
+  iSetup.get<HcalElectronicsMapRcd>().get(electronicsMapLabel_,item);
+  const HcalElectronicsMap* readoutMap = item.product();
+
+  //collection to be inserted into event
+  std::unique_ptr<FEDRawDataCollection> fed_buffers(new FEDRawDataCollection());
+  
+  //
+  //  Extracting All the Collections containing useful Info
+  iEvent.getByToken(tok_QIE10DigiCollection_,qie10DigiCollection);
+  const QIE10DigiCollection& qie10dc=*(qie10DigiCollection);
+  iEvent.getByToken(tok_QIE11DigiCollection_,qie11DigiCollection);
+  const QIE11DigiCollection& qie11dc=*(qie11DigiCollection);
+  iEvent.getByToken(tok_HBHEDigiCollection_,hbheDigiCollection);
+  const HBHEDigiCollection& qie8hbhedc=*(hbheDigiCollection);
+  iEvent.getByToken(tok_HFDigiCollection_,hfDigiCollection);
+  const HFDigiCollection& qie8hfdc=*(hfDigiCollection);
+  iEvent.getByToken(tok_TPDigiCollection_,tpDigiCollection);
+  const HcalTrigPrimDigiCollection& qietpdc=*(tpDigiCollection);
```

Comment:
There are no checks of validity of QIE10, QIE11, QIE8 (HF and HBHE) as well as TP digi collections.
Is it safe to assume that digitizers for all will be available?
I think it's better to prepare to pack only what's available instead of expecting that upstream will provide all types of collections here.